src/runner/bug_report.cpp

```diff
@@ -1,33 +1,300 @@
 
     bool expected_isBugReportThreadRunning = false;
     if (isBugReportThreadRunning.compare_exchange_strong(expected_isBugReportThreadRunning, true))
     {
-        std::thread([bug_report_path]() {
-            SHELLEXECUTEINFOW sei{ sizeof(sei) };
-            sei.fMask = { SEE_MASK_FLAG_NO_UI | SEE_MASK_NOASYNC | SEE_MASK_NOCLOSEPROCESS | SEE_MASK_NO_CONSOLE };
-            sei.lpFile = bug_report_path.c_str();
-            sei.nShow = SW_HIDE;
-            if (ShellExecuteExW(&sei))
+        SHELLEXECUTEINFOW sei{ sizeof(sei) };
+        sei.fMask = { SEE_MASK_FLAG_NO_UI | SEE_MASK_NOASYNC | SEE_MASK_NOCLOSEPROCESS | SEE_MASK_NO_CONSOLE };
+        sei.lpFile = bug_report_path.c_str();
+        sei.nShow = SW_HIDE;
+        if (ShellExecuteExW(&sei))
+        {
+            while (WaitForSingleObject(sei.hProcess, 100) == WAIT_TIMEOUT)
+            {
+                if (canceled.load())
+                {
+                    TerminateProcess(sei.hProcess, 0);
+                    CloseHandle(sei.hProcess);
+                    isBugReportThreadRunning.store(false);
+                    return "";
+                }
+            }
+            CloseHandle(sei.hProcess);
+
+            // Find the newest bug report file on the desktop
+            bugReportFileName = FindNewestBugReportFile();
+        }
+
+        isBugReportThreadRunning.store(false);
+    }
+    return bugReportFileName;
+}
+
+void InitializeReportBugLinkAsync()
+{
+    std::string gitHubURL;
+    std::string bugReportResult;
+
+    notifications::show_toast(GET_RESOURCE_STRING(IDS_BUGREPORT_TEXT), GET_RESOURCE_STRING(IDS_BUGREPORT_TITLE));
+
+    // Launch the bug report task
+    auto bugReportTask = std::async(std::launch::async, [&bugReportResult] {
+        bugReportResult = LaunchBugReport();
+    });
+
+    bugReportTask.wait();
+    
+    if (!bugReportResult.empty())
+    {
+        std::wstring wVersion = get_product_version();
+        std::string version;
+        std::transform(wVersion.begin() + 1, wVersion.end(), std::back_inserter(version), [](wchar_t c) {
+            return static_cast<char>(c);
+        });
+
+        std::string additionalInfo = "OS Build Version: " + GetOSVersion() + "%0a" + ".NET Version: " + GetDotNetVersion() + "%0a%0a";
+        GeneralSettings generalSettings = get_general_settings();
+        std::string isElevatedRun = generalSettings.isElevated ? "Running as admin: Yes" : "Running as admin: No";
+
+        std::string windowsSettings = ReportWindowsSettings();
+
+        const InstallScope current_install_scope = get_current_install_scope();
+
+        std::string installScope = current_install_scope == InstallScope::PerUser ? "Installation : User" : "Installation : System";
+
+        additionalInfo += windowsSettings + "%0a" + installScope + "%0a" + isElevatedRun;
+
+        gitHubURL = "https://github.com/microsoft/PowerToys/issues/new?assignees=&labels=Issue-Bug%2CNeeds-Triage&template=bug_report.yml" +
```

Comment:
Why assigning labels this way. They are defined in the template. Worry that nobody imagine about this piece of code later.

Reply:
https://aka.ms/powerToysReportBug
When you open this currently working link, the link that opens is as follows. Also, it does not assign these labels when they are not added.

https://github.com/microsoft/PowerToys/issues/new?assignees=&labels=Issue-Bug%2CTriage-Needed&template=bug_report.yml&title=